Source/Core/Common/Logging/Log.h

```diff
@@ -139,6 +142,13 @@ void GenericLog(LogLevel level, LogType type, const char* file, int line, const
     GENERIC_LOG(Common::Log::LogType::t, Common::Log::LogLevel::LDEBUG, __VA_ARGS__);              \
   } while (0)
 
+#define GENERIC_LOG_V(t, v, fmt, args)                                                             \
+  do                                                                                               \
```

Comment:
Just curious, why do these macros have a `do...while` loop?

Reply:
It makes a macro behave like a regular statement. In particular, think about how the macro would behave if placed in a non-braced if/for/while statement. This also makes it require the semicolon at the end at the usage site.

Reply:
https://www.quora.com/What-is-the-purpose-of-using-do-while-0-in-macros and https://stackoverflow.com/a/154138 talk about it and seem like OK explanations - it's basically so that the macro is treated as a single statement and doesn't behave strangely with unbraced if statements.  I don't think they're actually necessary here, but it's a good habit and there's no harm in using it everywhere.

Reply:
^ I think even here you could break this if you didn't use a `do { } while (0)` with something like:

```
if (condition)
  GENERIC_LOG(loglevel, whatever);
else
   something_else();
```

The else would bind to the GENERIC_LOG's inner if, and not the outer `if (condition)`.

Reply:
The ones on `ERROR_LOG`/`WARN_LOG`/etc might be redundant though since they're a single function; I think this would also work without issues (as long as `GENERIC_LOG` still had do-while):

```C++
#define ERROR_LOG(t, ...)                                                                          \
  GENERIC_LOG(Common::Log::LogType::t, Common::Log::LogLevel::LERROR, __VA_ARGS__)
```

Note the lack of a trailing semicolon.

Reply:
Thanks for the details everyone.  Now that you mention it @Pokechu22 , I do recall seeing that so post.  Good stuff!